recorder_test.go

```diff
@@ -26,16 +26,18 @@ func TestRecorder(t *testing.T) {
 	r := NewRecorder(id, c)
 
 	r.Msg("msg")
-	if calledCollect != 1 {
-		t.Errorf("got calledCollect %d, want 1", calledCollect)
+	if calledCollect != 0 {
+		t.Errorf("got calledCollect %d, want 0", calledCollect)
 	}
+	r.Finish()
 	if diff := diffAnnotationsFromEvent(anns, Msg("msg")); len(diff) > 0 {
 		t.Errorf("got diff annotations for Msg event:\n%s", strings.Join(diff, "\n"))
 	}
 
 	r.Name("name")
+	r.Finish()
 	if calledCollect != 2 {
-		t.Errorf("got calledCollect %d, want 1", calledCollect)
+		t.Errorf("got calledCollect %d, want 2", calledCollect)
 	}
```

Comment:
Finish should never be called twice, so we shouldn't test for this. Instead, let's make Finish being called twice an error by printing a log message from within `Finish` like:

```
log.Println("appdash: multiple Recorder.Finish calls")
```

and then have it silently make the call a no-op.